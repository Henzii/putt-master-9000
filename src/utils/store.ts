import { createStore, combineReducers } from "redux";
import gameDataReducer from '../reducers/gameDataReducer';
import notificationReducer from '../reducers/notificationReducer';
import selectedLayoutReducer from '../reducers/selectedLayoutReducer';
import commonReducer from '../reducers/commonReducer';
import userReducer from '../reducers/userReducer';

const reducers = combineReducers({
    gameData: gameDataReducer,
    notifications: notificationReducer,
    selectedLayout: selectedLayoutReducer,
    user: userReducer,
    common: commonReducer
});

export type RootState = ReturnType<typeof reducers>;

export default createStore(reducers);
