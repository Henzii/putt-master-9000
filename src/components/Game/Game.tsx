import { View, Text, StyleSheet } from 'react-native';
import React, { useState } from 'react';
import Player from './Player';
import RoundTabs from './RoundTabs';
import { theme } from '../../utils/theme';
import useGame from '../../hooks/useGame';
import Container from '../ThemedComponents/Container';
import { Button, Paragraph, Title } from 'react-native-paper';
import { useDispatch } from 'react-redux';
import { unloadGame } from '../../reducers/gameDataReducer';

export default function Game({ gameId }: { gameId: string }) {
    const [selectedRound, setSelectedRound] = useState(0);
    const { data, ready, error, setScore } = useGame(gameId);
    const handleScoreChange = (playerId: string, selectedRound: number, value: number) => {
        setScore({
            gameId,
            playerId,
            hole: selectedRound,
            value,
        })
    }
    if (error?.message) {
        return (
            <Container>
                <Title>Error!</Title>
                <Paragraph>
                    {error.message}
                </Paragraph>
            </Container>
        )
    }
    if (!data || !ready) {
        return (
            <View>
                <Text>Loading or something....</Text>
            </View>
        )
    }
    if (!data.isOpen) {
        return <ClosedGame />
    }
    return (
        <>
            <RoundTabs gameData={data} selectedRound={selectedRound} setSelectedRound={setSelectedRound} />
            <Container noPadding>
                <View style={peliStyles.headers}>
                    <Text style={peliStyles.course}>{data.course} #{selectedRound + 1}, par {data.pars[selectedRound]}</Text>
                    <Text style={peliStyles.layout}>{data.layout}</Text>
                </View>
                {data.scorecards.map(p =>
                    <Player
                        pars={data.pars}
                        key={p.user.id}
                        player={p}
                        selectedRound={selectedRound}
                        setScore={handleScoreChange}
                    />
                )}
            </Container>
        </>
    )
}
const ClosedGame = () => {
    const dispatch = useDispatch();

    // Uusi peli -> poistetaan pelin tiedot redux-storesta jolloin createGame näkymä avautuu
    const handleButtonClick = () => {
        dispatch(unloadGame());
    }
    return (
        <View style={peliStyles.gameover}>
            <Title style={peliStyles.gameoverText}>Game over!</Title>
            <Button onPress={handleButtonClick}>Start a new game</Button>
        </View>
    )
}
const peliStyles = StyleSheet.create({
    headers: {
        padding: 10,
    },
    gameover: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        
    },
    gameoverText: {
        color: 'gray',
        fontSize: 30,
    },
    course: {
        fontSize: theme.font.sizes.huge + 5,
        fontWeight: 'bold',
    },
    layout: {
        color: 'gray',
    }
})