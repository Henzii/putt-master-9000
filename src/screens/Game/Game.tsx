import { View, Text, StyleSheet, FlatList } from 'react-native';
import React, { useState } from 'react';
import Player from './Player';
import { Dimensions } from 'react-native';
import RoundTabs from '../../components/RoundTabs';
import useGame from '../../hooks/useGame';
import Container from '../../components/ThemedComponents/Container';
import { Button, Title } from 'react-native-paper';
import { useDispatch, useSelector } from 'react-redux';
import { gameData, unloadGame } from '../../reducers/gameDataReducer';
import Loading from '../../components/Loading';
import { RootState } from '../../utils/store';
import ErrorScreen from '../../components/ErrorScreen';
import { useCallback } from 'react';

export default function Game() {
    const [selectedRound, setSelectedRound] = useState(0);
    const gameData = useSelector((state: RootState) => state.gameData) as gameData;
    const gameId = gameData.gameId;
    const { data, loading, error, setScore } = useGame(gameId);

    const handleScoreChange = (playerId: string, selectedRound: number, value: number) => {
        setScore({
            gameId,
            playerId,
            hole: selectedRound,
            value,
        });
    };
    const findThrowingOrder = useCallback(() => {
        // Kopioidaan tuloskortit
        const cards = [...data?.scorecards || []];
        for (let i = 1; i < selectedRound; i++) {
            cards.sort((a, b) => (a.scores[i] || 99) - (b.scores[i] || 99));
        }
        return cards.reduce((p,c,i) => {
            p[c.user.id] = (i+1);
            return p;
        }, ({} as { [key: string]: number }));
    }, [selectedRound]);

    if (!data && loading) return <Loading />;
    if (!data || error) {
        return <ErrorScreen errorMessage={`Error just happened!`} />;
    }
    if (!data.isOpen) {
        return <ClosedGame />;
    }
    // Apumuuttuja jolla todetaan swiippaus vasemmalle
    let touchPos = [0, 0];
    const throwingOrder = findThrowingOrder();
    return (
        <>
            <RoundTabs gameData={data} selectedRound={selectedRound} setSelectedRound={setSelectedRound} />
            <Container noPadding>
                <View
                    style={{ minHeight: '100%' }}
                    onTouchStart={(e) => touchPos = [e.nativeEvent.pageX, e.nativeEvent.pageY]}
                    onTouchEnd={(e) => {
                        // 50 % ruudun leveydesä
                        const width50 = Dimensions.get('screen').width * 0.5;
                        // Jos kosketus on Y-suunnassa yli 30 -> return
                        if (Math.abs(e.nativeEvent.pageY - touchPos[1]) >= 50) return;

                        const movement = e.nativeEvent.pageX - touchPos[0];
                        // Jos liike X-suunnassa on vähemmän kuin 70% leveydestä
                        if (Math.abs(movement) < width50) return;

                        if (movement > 0 && selectedRound > 0) setSelectedRound((v) => v - 1);
                        else if (movement < 0 && selectedRound < data.holes - 1) setSelectedRound((v) => v + 1);
                    }}
                >
                    <View
                        style={peliStyles.headers}
                    >
                        <Text testID="GameRata" style={peliStyles.course}>{data.course} #{selectedRound + 1}, par {data.pars[selectedRound]}</Text>
                        <Text testID="GameLayout" style={peliStyles.layout}>{data.layout}</Text>
                    </View>
                    <FlatList
                        data={data.scorecards}
                        keyExtractor={(item) => item.user.id as string}
                        ListFooterComponent={<View style={{ height: 100 }} />}
                        renderItem={({ item }) => (
                            <Player
                                player={item}
                                selectedRound={selectedRound}
                                setScore={handleScoreChange}
                                order={throwingOrder[item.user.id]}
                            />
                        )}
                    />
                </View>
            </Container>
        </>
    );
}
const ClosedGame = () => {
    const dispatch = useDispatch();

    // Uusi peli -> poistetaan pelin tiedot redux-storesta jolloin createGame näkymä avautuu
    const handleButtonClick = () => {
        dispatch(unloadGame());
    };
    return (
        <Container style={peliStyles.gameover}>
            <Title style={peliStyles.gameoverText}>Game over!</Title>
            <Button onPress={handleButtonClick}>Start a new game</Button>
        </Container>
    );
};
const peliStyles = StyleSheet.create({
    headers: {
        padding: 10,
    },
    gameover: {
        justifyContent: 'center',
        alignItems: 'center',

    },
    gameoverText: {
        color: 'gray',
        fontSize: 30,
    },
    course: {
        fontSize: 30,
        fontWeight: 'bold',
    },
    layout: {
        color: 'gray',
    }
});