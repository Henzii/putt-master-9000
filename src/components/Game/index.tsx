import React, { useState } from 'react';
import { useMutation } from 'react-apollo';
import { BottomNavigation } from 'react-native-paper';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-native';
import { ADD_PLAYERS_TO_GAME, CREATE_GAME } from '../../graphql/mutation';
import { GET_OLD_GAMES } from '../../graphql/queries';
import { gameData, newGame } from '../../reducers/gameDataReducer';
import { addNotification } from '../../reducers/notificationReducer';
import { RootState } from '../../utils/store';
import CreateGame, { NewGameData } from './CreateGame';
import Game from './Game';
import Setup from './Setup';
import Summary from './Summary';

export default function() {
    const gameData = useSelector((state: RootState) => state.gameData) as gameData;
    const [createGameMutation] = useMutation(CREATE_GAME, { refetchQueries: [ { query: GET_OLD_GAMES }]});
    const [addPlayersMutation] = useMutation(ADD_PLAYERS_TO_GAME);
    const dispatch = useDispatch();
    const navi = useNavigate();

    // Alanaville:
    const [navIndex, setNavIndex] = useState(0);
    const [navRoutes] = useState([
        {key: 'gameRoute', title: 'Scorecard', icon: 'counter'},
        {key: 'summaryRoute', title: 'Summary', icon: 'format-list-numbered'},
        {key: 'setupRoute', title: 'Setup', icon: 'cog-outline'},
    ])
    const naviScenes = BottomNavigation.SceneMap({
        gameRoute: () => <Game gameId={gameData.gameId} />,
        setupRoute: Setup,
        summaryRoute: Summary,
    })
    const handleCreateGame = async (data: NewGameData) => {
        if (!data.layout || !data.course) return;
        const res = await createGameMutation({
            variables: { courseId: data.course.id, layoutId: data.layout.id }
        })
        const newGameId = res.data.createGame;
        console.log('New game:', newGameId);
        await addPlayersMutation({ variables: {
            gameId: newGameId,
            playerIds: data.players.map(p => p.id)
        }})
        dispatch(newGame(newGameId));
        dispatch(addNotification('New game created!', 'success'));
    }
    // Jos peli채 ei ladattuna, n채ytet채채n CreateGame
    if (!gameData?.gameId) {
        return <CreateGame onCreate={handleCreateGame} onCancel={() => navi(-1)} />
    }
    return (
        <BottomNavigation
            style={{ width: '100%' }}
            navigationState={{ index: navIndex, routes: navRoutes}}
            onIndexChange={setNavIndex}
            renderScene={naviScenes}
        />
    )    
    
}

