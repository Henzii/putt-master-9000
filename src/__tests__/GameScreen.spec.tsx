import React from 'react';

import { render, waitFor } from '@testing-library/react-native';
import Game from '../screens/Game/Game';
import { MockedProvider } from '@apollo/react-testing';
import { Provider as ReduxProvider } from 'react-redux';
import store from '../utils/store';
//import { Provider } from 'react-native-paper';
import * as gameMocks from './mocks/gameMocks';
import { InMemoryCache } from 'apollo-boost';

const wrappedGame = () => {
    return (
        <ReduxProvider store={store}>
            <MockedProvider mocks={[gameMocks.mockedQuery]} addTypename={true}
                cache={
                    new InMemoryCache({
                        addTypename: false,
                        fragmentMatcher: { match: () => true },
                    })
                }
            >
                <Game gameId='mockedGame1' />
            </MockedProvider>
        </ReduxProvider>
    );
};

describe('<Game /> testit', () => {
    it('Alussa loading... teksti', async () => {
        const { getByTestId } = render(wrappedGame());
        expect(getByTestId('progress')).toBeDefined();
    });
    it('Radan nimi ja pelaajat löytyvät', async () => {
        const { getByTestId, getByText } = render(wrappedGame());
        await waitFor(() => {

            // Radan nimi löytyy...
            expect(getByTestId('GameRata').children).toContain(gameMocks.mockedGame.course);

            // Molemmat pelaajat löytyy
            expect(getByText(gameMocks.mockedGame.scorecards[0].user.name)).not.toBeNull();
        });
    });
});