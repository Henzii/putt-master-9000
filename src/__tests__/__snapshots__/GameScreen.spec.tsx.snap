// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game /> testit Render√∂ityy oikein 1`] = `
Array [
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
      horizontal={true}
      showsHorizontalScrollIndicator={false}
    >
      <View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="SingleTab"
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgb(220,220,220)",
                  "borderColor": "lightgray",
                  "borderTopLeftRadius": 7,
                  "borderTopRightRadius": 7,
                  "borderWidth": 1,
                  "height": 50,
                  "paddingTop": "20%",
                  "width": 50,
                },
                Object {
                  "backgroundColor": "rgb(180, 255, 180)",
                  "borderBottomWidth": 0,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              1
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="SingleTab"
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgb(220,220,220)",
                  "borderColor": "lightgray",
                  "borderTopLeftRadius": 7,
                  "borderTopRightRadius": 7,
                  "borderWidth": 1,
                  "height": 50,
                  "paddingTop": "20%",
                  "width": 50,
                },
                Object {
                  "backgroundColor": "rgb(180,220,180)",
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              2
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="SingleTab"
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgb(220,220,220)",
                  "borderColor": "lightgray",
                  "borderTopLeftRadius": 7,
                  "borderTopRightRadius": 7,
                  "borderWidth": 1,
                  "height": 50,
                  "paddingTop": "20%",
                  "width": 50,
                },
                Object {
                  "backgroundColor": "rgb(180,220,180)",
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              3
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="SingleTab"
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgb(220,220,220)",
                  "borderColor": "lightgray",
                  "borderTopLeftRadius": 7,
                  "borderTopRightRadius": 7,
                  "borderWidth": 1,
                  "height": 50,
                  "paddingTop": "20%",
                  "width": 50,
                },
                Object {
                  "backgroundColor": "rgb(180,220,180)",
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              4
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="SingleTab"
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgb(220,220,220)",
                  "borderColor": "lightgray",
                  "borderTopLeftRadius": 7,
                  "borderTopRightRadius": 7,
                  "borderWidth": 1,
                  "height": 50,
                  "paddingTop": "20%",
                  "width": 50,
                },
                Object {
                  "backgroundColor": "rgb(180,220,180)",
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              5
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>,
  <View
    style={
      Array [
        undefined,
        undefined,
        undefined,
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#f6f6f6",
          "padding": 0,
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "minHeight": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 30,
              "fontWeight": "bold",
            }
          }
          testID="GameRata"
        >
          MockedCourse1
           #
          1
          , par 
          3
        </Text>
        <Text
          style={
            Object {
              "color": "gray",
            }
          }
          testID="GameLayout"
        >
          MockedLayout1
        </Text>
      </View>
      <RCTScrollView
        ItemSeparatorComponent={[Function]}
        ListFooterComponent={
          <View
            style={
              Object {
                "height": 70,
              }
            }
          />
        }
        data={
          Array [
            Object {
              "beers": 0,
              "hc": 0,
              "plusminus": 2,
              "scores": Array [
                3,
                3,
                3,
                3,
                3,
              ],
              "total": 15,
              "user": Object {
                "id": "MockedUserId1",
                "name": "MockerUser1",
              },
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "flex": 1,
          }
        }
        viewabilityConfigCallbackPairs={Array []}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              collapsable={false}
              onLayout={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "#ffffff",
                  "borderColor": "rgba(0, 0, 0, 0.12)",
                  "borderRadius": 4,
                  "elevation": 2,
                  "marginBottom": 1,
                  "minHeight": 120,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 0.75,
                    "width": 0,
                  },
                  "shadowOpacity": 0.24,
                  "shadowRadius": 1.5,
                  "width": "98%",
                }
              }
            >
              <View
                accessibilityState={
                  Object {
                    "disabled": true,
                  }
                }
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flexGrow": 1,
                    "flexShrink": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "paddingLeft": 16,
                      },
                      Object {
                        "minHeight": 72,
                      },
                      Object {
                        "padding": 10,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "flexDirection": "column",
                          "justifyContent": "center",
                        },
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "textAlign": "left",
                            },
                            Object {
                              "color": "rgba(0, 0, 0, 0.87)",
                              "fontFamily": "System",
                              "fontWeight": "500",
                              "writingDirection": "ltr",
                            },
                            Array [
                              Object {
                                "fontSize": 20,
                                "letterSpacing": 0.15,
                                "lineHeight": 30,
                                "marginVertical": 2,
                              },
                              Array [
                                Object {
                                  "minHeight": 30,
                                  "paddingRight": 16,
                                },
                                Object {
                                  "marginBottom": 2,
                                },
                                undefined,
                              ],
                            ],
                          ],
                        ]
                      }
                    >
                      MockerUser1
                    </Text>
                  </View>
                  <View>
                    <Text
                      style={
                        Object {
                          "fontSize": 12,
                          "opacity": 0.6,
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "paddingHorizontal": 16,
                      },
                      undefined,
                      Object {
                        "display": "flex",
                        "flexDirection": "row",
                        "justifyContent": "space-around",
                        "padding": 10,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 4,
                      }
                    }
                  >
                    <RCTScrollView
                      data={
                        Array [
                          1,
                          2,
                          3,
                          4,
                          5,
                          6,
                          7,
                          8,
                          9,
                        ]
                      }
                      getItem={[Function]}
                      getItemCount={[Function]}
                      horizontal={true}
                      initialScrollIndex={0}
                      keyExtractor={[Function]}
                      onContentSizeChange={[Function]}
                      onEndReached={[Function]}
                      onEndReachedThreshold={0.1}
                      onLayout={[Function]}
                      onMomentumScrollBegin={[Function]}
                      onMomentumScrollEnd={[Function]}
                      onScroll={[Function]}
                      onScrollBeginDrag={[Function]}
                      onScrollEndDrag={[Function]}
                      removeClippedSubviews={false}
                      renderItem={[Function]}
                      scrollEventThrottle={50}
                      stickyHeaderIndices={Array []}
                      viewabilityConfigCallbackPairs={Array []}
                    >
                      <View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                              null,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "lightgray",
                                  "borderColor": "#000",
                                  "borderRadius": 7,
                                  "borderWidth": 1,
                                  "elevation": 2,
                                  "marginRight": 5,
                                  "paddingHorizontal": 15,
                                  "paddingVertical": 10,
                                },
                                false,
                                false,
                              ]
                            }
                          >
                            <Text>
                              1
                            </Text>
                          </View>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                              null,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "lightgray",
                                  "borderColor": "#000",
                                  "borderRadius": 7,
                                  "borderWidth": 1,
                                  "elevation": 2,
                                  "marginRight": 5,
                                  "paddingHorizontal": 15,
                                  "paddingVertical": 10,
                                },
                                false,
                                false,
                              ]
                            }
                          >
                            <Text>
                              2
                            </Text>
                          </View>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                              null,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "lightgray",
                                  "borderColor": "#000",
                                  "borderRadius": 7,
                                  "borderWidth": 1,
                                  "elevation": 2,
                                  "marginRight": 5,
                                  "paddingHorizontal": 15,
                                  "paddingVertical": 10,
                                },
                                false,
                                Object {
                                  "backgroundColor": "#8ecf8a",
                                  "borderColor": "green",
                                },
                              ]
                            }
                          >
                            <Text>
                              3
                            </Text>
                          </View>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                              null,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "lightgray",
                                  "borderColor": "#000",
                                  "borderRadius": 7,
                                  "borderWidth": 1,
                                  "elevation": 2,
                                  "marginRight": 5,
                                  "paddingHorizontal": 15,
                                  "paddingVertical": 10,
                                },
                                false,
                                false,
                              ]
                            }
                          >
                            <Text>
                              4
                            </Text>
                          </View>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                              null,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "lightgray",
                                  "borderColor": "#000",
                                  "borderRadius": 7,
                                  "borderWidth": 1,
                                  "elevation": 2,
                                  "marginRight": 5,
                                  "paddingHorizontal": 15,
                                  "paddingVertical": 10,
                                },
                                false,
                                false,
                              ]
                            }
                          >
                            <Text>
                              5
                            </Text>
                          </View>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                              null,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "lightgray",
                                  "borderColor": "#000",
                                  "borderRadius": 7,
                                  "borderWidth": 1,
                                  "elevation": 2,
                                  "marginRight": 5,
                                  "paddingHorizontal": 15,
                                  "paddingVertical": 10,
                                },
                                false,
                                false,
                              ]
                            }
                          >
                            <Text>
                              6
                            </Text>
                          </View>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                              null,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "lightgray",
                                  "borderColor": "#000",
                                  "borderRadius": 7,
                                  "borderWidth": 1,
                                  "elevation": 2,
                                  "marginRight": 5,
                                  "paddingHorizontal": 15,
                                  "paddingVertical": 10,
                                },
                                false,
                                false,
                              ]
                            }
                          >
                            <Text>
                              7
                            </Text>
                          </View>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                              null,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "lightgray",
                                  "borderColor": "#000",
                                  "borderRadius": 7,
                                  "borderWidth": 1,
                                  "elevation": 2,
                                  "marginRight": 5,
                                  "paddingHorizontal": 15,
                                  "paddingVertical": 10,
                                },
                                false,
                                false,
                              ]
                            }
                          >
                            <Text>
                              8
                            </Text>
                          </View>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                              null,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "lightgray",
                                  "borderColor": "#000",
                                  "borderRadius": 7,
                                  "borderWidth": 1,
                                  "elevation": 2,
                                  "marginRight": 5,
                                  "paddingHorizontal": 15,
                                  "paddingVertical": 10,
                                },
                                false,
                                false,
                              ]
                            }
                          >
                            <Text>
                              9
                            </Text>
                          </View>
                        </View>
                      </View>
                    </RCTScrollView>
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "fontSize": 23,
                        }
                      }
                    >
                      +
                      2
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
          >
            <View
              style={
                Object {
                  "height": 70,
                }
              }
            />
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>,
]
`;
